### ISSUE TITLE
File name mistype detection

### ISSUE THREAD

**Issue by Echua36** [godotengine/godot-proposals#1748]
**Describe the project you are working on:**
i'm working on a game, build for intel/low end computer
**Describe the problem or limitation you are having in your project:**
there's no limitation to my project, but there's typing error problem
this is the bug report:https://github.com/godotengine/godot/issues/43172, as you can see i got my files named wrongly, and i can't founds out what's wrong without someone pointing out "You've got the names for your scenes wrong."
**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
this is good because, easier to debug the wrong name, and less "why my code won't work?"
and there's already "if node existed". and why not add "if file existed" too
![Godot_v3 2 3-stable_win64_nHim7PJLKQ](https://user-images.githubusercontent.com/45130922/97606392-b2869a80-1a4a-11eb-82e2-26cb76f46ef4.png)
**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
i don't understand the question
**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
it can work around by delete the 'wrong name' and drag the correct resouce in the Gdscript it, but i found tedious doing it
and, i don't even know the file's name are wrong in first place
**Is there a reason why this should be core and not an add-on in the asset library?:**
the reason is improve the Gdscript's type error detection for everyone

**Comment by Calinou** [comment]
`load()` is only interpreted at run-time so that you can use a non-constant expression inside. We can't know if the file is missing before the engine tries to load it.

**Comment by Echua36** [comment]
but you can add "res://1kBills Tree_16_1.tscn" is missing after the engine tries to load it, but failed as a Warnings

**Comment by mrjustaguy** [comment]
Yeah, adding some extra data to the errors (like parameter values it tried to run when it failed) would be a nice feature.. how worth it is it compared to the time it takes to implement something like that and how exactly one would implement it... hard to say.
